### ISSUE TITLE
filter function can throw uncatchable errors

### ISSUE THREAD

**Issue by marten-de-vries** [pouchdb/pouchdb#3356]
```javascript
var PouchDB = require('pouchdb');
var db = new PouchDB('test', {db: require('memdown')});
db.put({
  _id: "_design/test",
  filters: {
    test: function (doc, req) {
      throw new Error(); // syntaxerrors can't be caught either.
    }.toString()
  }
})
db.changes({filter: 'test/test'}).catch(function (err) {
  console.log("this statement is never reached although an error is thrown.")
});
```

Why is this bad? Because it crashes express-pouchdb. For bonus points, also known as skipping less of couchdb-harness, making the second argument of a filter function ``req`` specifiable as an option to ``changes()`` would be nice.

**Comment by nolanlawson** [comment]
Ouch. Same thing happens with map/reduce functions btw.

**Comment by nolanlawson** [comment]
fixed by 12cfc598402f0aebd3df8c669daf601ab9ec5086
